When I try to create ath_masker, I get an error: struct kprobe’ has no member named ‘fault_handler’;

cd wifite2                                                                                                                                                
┌──(user1㉿kali)-[~/wifite2]
└─$ make ath_masker               
cd tools/ath_masker/ && ./load.sh
make[1]: directory entry «/home/user1/wifite2/tools/ath_masker»
make -C /lib/modules/5.14.0-kali4-amd64/build  M=/home/user1/wifite2/tools/ath_masker
make[2]: directory entry «/usr/src/linux-headers-5.14.0-kali4-amd64»
  CC [M]  /home/user1/wifite2/tools/ath_masker/ath_masker.o
/home/user1/wifite2/tools/ath_masker/ath_masker.c: In function ‘kprobe_init’:
/home/user1/wifite2/tools/ath_masker/ath_masker.c:95:5: error: ‘struct kprobe’ has no member named ‘fault_handler’; did you mean ‘post_handler’?
   95 |  kp.fault_handler = handler_fault;

--- a/tools/ath_masker/ath_masker.c
+++ b/tools/ath_masker/ath_masker.c
@@ -50,6 +50,8 @@ static int handler_pre(struct kprobe *p, struct pt_regs *regs)
 	struct ath_common *common = (struct ath_common *)regs->ax;
 #elif defined(CONFIG_ARM)
 	struct ath_common *common = (struct ath_common *)regs->ARM_r0;
+#elif defined(CONFIG_ARM64)
+  struct ath_common *common = (struct ath_common *)regs->regs[0];
 #endif
 
 	printk("pre_handler: MAC address of device is %pM\n", common->macaddr);
@@ -68,31 +70,17 @@ static void handler_post(struct kprobe *p, struct pt_regs *regs,
 	// Nothing
 }
 
-/*
- * fault_handler: this is called if an exception is generated for any
- * instruction within the pre- or post-handler, or when Kprobes
- * single-steps the probed instruction.
- */
-static int handler_fault(struct kprobe *p, struct pt_regs *regs, int trapnr)
-{
-	printk(KERN_INFO "fault_handler: p->addr = 0x%p, trap #%dn",
-		p->addr, trapnr);
-	/* Return 0 because we don't handle the fault. */
-	return 0;
-}
-
 static int __init kprobe_init(void)
 {
 	int ret;
 
-#if !defined(CONFIG_X86_64) && !defined(CONFIG_X86) && !defined(CONFIG_ARM)
-	printk(KERN_ALERT "Error: this module only supports x86(_64) or ARM\n");
+#if !defined(CONFIG_X86_64) && !defined(CONFIG_X86) && !defined(CONFIG_ARM) && !defined(CONFIG_ARM64)
+	printk(KERN_ALERT "Error: this module only supports x86(_64) or ARM(64)\n");
 	return -EINVAL;
 #endif
 
 	kp.pre_handler = handler_pre;
 	kp.post_handler = handler_post;
-	kp.fault_handler = handler_fault;
 
 	ret = register_kprobe(&kp);
 	if (ret < 0) {

